src/validatefs/validatefs.c

```diff
@@ -119,46 +119,72 @@ static int parse_argv(int argc, char *argv[]) {
 }
 
 typedef struct ValidateFields {
-        sd_id128_t gpt_type_uuid;
-        char *gpt_label;
+        sd_id128_t *gpt_type_uuid;
+        size_t n_gpt_type_uuid;
+        char **gpt_label;
         char **mount_point;
 } ValidateFields;
 
 static void validate_fields_done(ValidateFields *f) {
         assert(f);
 
-        free(f->gpt_label);
+        free(f->gpt_type_uuid);
+        strv_free(f->gpt_label);
         strv_free(f->mount_point);
 }
 
+static char *validate_fields_gpt_type_uuid_as_string(const ValidateFields *f) {
+        _cleanup_free_ char *joined = NULL;
+        assert(f);
+
+        FOREACH_ARRAY(u, f->gpt_type_uuid, f->n_gpt_type_uuid)
+                if (!strextend_with_separator(&joined, ", ", SD_ID128_TO_UUID_STRING(*u)))
```

Comment:
hmm, this is pre-existing, but should we also log about pretty names here, i.e. gpt_partition_type_uuid_to_string?